source/ui/dataman/AMScanThumbnailGridViewItemDelegate.cpp

```diff
@@ -0,0 +1,92 @@
+#include "AMScanThumbnailGridViewItemDelegate.h"
+#include "util/AMDateTimeUtils.h"
+
+AMScanThumbnailGridViewItemDelegate::AMScanThumbnailGridViewItemDelegate(QObject *parent) :
+	QItemDelegate(parent)
+{
+}
+
+void AMScanThumbnailGridViewItemDelegate::paint(QPainter *painter,
+												const QStyleOptionViewItem &option,
+												const QModelIndex &index) const
+{
+
+	if(option.rect.isEmpty() || !index.isValid())
+		return;
+
+	painter->save();
+
+	if(!index.parent().isValid()) {
+		// Is a scan
+
+		switch (index.column()) {
+		case 0:
+			paintItemBackground(painter, option);
+			break;
+		case 1:
+			paintText(painter, option, index.data(Qt::DisplayRole).toString());
+			break;
+		case 3:
+			paintText(painter, option,
+					  AMDateTimeUtils::prettyDateTime(index.data(Qt::DisplayRole).toDateTime()));
+			break;
+		case 4:
+			paintText(painter, option, index.data(Qt::DisplayRole).toString());
+			break;
+		default:
+			break;
+		}
+
+	} else {
+		// Is a thumbnail
+		switch (index.column()) {
+		case 1:
+			paintThumbnail(painter, option, index.data(Qt::DecorationRole).value<QPixmap>());
+			break;
+		case 2:
+			paintText(painter, option, index.data(Qt::DisplayRole).toString());
+			break;
+		default:
+			break;
+		}
+	}
+
+	painter->restore();
+}
+
+void AMScanThumbnailGridViewItemDelegate::paintThumbnail(QPainter *painter,
+															  const QStyleOptionViewItem &option,
+															  const QPixmap &pixmap) const
+{
+	painter->drawPixmap(option.rect, pixmap);
```

Comment:
This is more of a curiosity question.  Does this method draw the pixmap within the confines of the provided QRect or does it draw what it can be chop extra or have dead space if the pixmap is smaller than the rect?